File: JumboModClient-Ios/JsonFetcher.m
//
//  MasterViewController.m
//  JumboModClient-Ios
//
//  Created by Aaron Evans on 11/15/14.
//  Copyright (c) 2014 Jumbo Mobile. All rights reserved.
//

//#import "MasterViewController.h"
#import "DetailViewController.h"

#import <Foundation/Foundation.h>
#import <RestKit/RestKit.h>

//#import "User.h"
#import "SplicerClient-Swift-Bridging-Header.h"
#import "JsonFetcher.h"

//static NSString * const SERVER_BASE_URL = @"http://splicer.io";
//static NSString * const SERVER_BASE_URL = @"http://localhost:9400";

@implementation JsonFetcher

//@synthesize detailViewController;
//@synthesize usersFetcher;
//@synthesize users;
NSArray *testUsers;
NSString * const SERVER_BASE_URL = @"http://localhost:9000";//http://localhost:9000/api/sports-roots
NSString * const AUTH_HEADER = @"Bearer eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJhZG1pbiIsImF1dGgiOiJST0xFX0FETUlOLFJPTEVfVVNFUiIsImV4cCI6MTU4ODYxNjY4NH0.Ln1M_VzMd_HXfR6J3EFDQfJudOBwqMf1JK32R3WlHZLvlf6wfgT9Cc7_I04Dotbi4Vp_EpHe0y4ZE9HVWQme0w";

//from orig jsonfetcher
@synthesize objectManager;
@synthesize mapping;
//@synthesize users;
//@synthesize tableView;


//below from orig jsonfetcher
-(void)setup:(NSString *)baseUrl rootClass:(Class)clazz path:(NSString *)path;
{
    //users = usrs;
    // initialize AFNetworking HTTPClient
    NSURL *baseURL = [NSURL URLWithString:baseUrl];//[NSURL URLWithString:@"https://api.foursquare.com"];
    AFRKHTTPClient *client = [[AFRKHTTPClient alloc] initWithBaseURL:baseURL];
//    [client setDefaultHeader:@"Accept" value:RKMIMETypeJSON];
    [client setDefaultHeader:@"Authorization" value:AUTH_HEADER];
    
    // initialize RestKit
    objectManager = [[RKObjectManager alloc] initWithHTTPClient:client];
    
    //map's aaron's original way
    RKObjectMapping* venueMapping = [self mapComplexTypes];
    
    // register mappings with the provider using a response descriptor
    RKResponseDescriptor *responseDescriptor =
    [RKResponseDescriptor responseDescriptorWithMapping:venueMapping
                                            method:RKRequestMethodGET
                                            pathPattern:path //@"/api/splicer/User/list"]
                                            //from sample: keyPath:@"response.venues"
                                            keyPath:@""
                                            statusCodes:[NSIndexSet indexSetWithIndex:200]];
    
    [objectManager addResponseDescriptor:responseDescriptor];
}

- (void)execute:(NSString *)path;
{
    //aaron's orig (0.1.0 or earlier)
    /*
     RKObjectLoader* loader = [objectManager objectLoaderWithResourcePath:path delegate:delegate];
     loader.objectMapping = mapping; // Here's the key. We're telling RestKit to use the MyItem mapping for the objects in the root.
     [loader send];
     */
    
    NSDictionary *queryParams = @{@"dummy":@"dummyParam"};
    
    [[RKObjectManager sharedManager] getObjectsAtPath:path//@"/v2/venues/search"
                                           parameters:queryParams
                                              success:^(RKObjectRequestOperation *operation, RKMappingResult *mappingResult) {
                                                  //FIXME also need a callback where this type gets created.  Or perhaps you can just typecast it
                                                  testUsers = mappingResult.array;
                                                  //FIXME: Need a way to invalidate the caller from this thread.
                                                  //[self.tableView reloadData];
                                                  
                                              }
                                              failure:^(RKObjectRequestOperation *operation, NSError *error) {
                                                  NSLog(@"Failure of GET:/user with error %@.", error);
                                              }];
    
}


@end

File: JumboModClient-Ios/JsonFetcher.swift
//
//  JsonFetcher.swift
//  SplicerClient-Swift
//
//  Created by Aaron Evans on 4/16/20.
//  Copyright © 2020 Aaron Evans. All rights reserved.
//

//  The converted code is limited to 2 KB.
//  Upgrade your plan to remove this limitation.
//
//  Converted to Swift 5.1 by Swiftify v5.1.31847 - https://swiftify.com/
import Foundation
import RestKit

var testUsers: [AnyHashable]?
let SERVER_BASE_URL = "http://localhost:9000" //
let AUTH_HEADER = "Bearer eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJhZG1pbiIsImF1dGgiOiJST0xFX0FETUlOLFJPTEVfVVNFUiIsImV4cCI6MTU4ODYxNjY4NH0.Ln1M_VzMd_HXfR6J3EFDQfJudOBwqMf1JK32R3WlHZLvlf6wfgT9Cc7_I04Dotbi4Vp_EpHe0y4ZE9HVWQme0w";

class JsonFetcher : JsonMapper {
    var testUsers: [Any]!
    
    var baseURL: URL!;// = URL(string: baseUrl ?? "")
    var client: AFRKHTTPClient!;// = AFRKHTTPClient(baseURL: baseURL)
    var objectManager: RKObjectManager!;// = RKObjectManager(httpClient: client)
    //var mapping: RKObjectMapping!;// = mapComplexTypes()
    var statusCodes: IndexSet = [];// = RKStatusCodeIndexSetForClass(RKStatusCodeClass.successful)
    var resDescriptor: RKResponseDescriptor!;// = RKResponseDescriptor(mapping: mapping, method: RKRequestMethod.GET, pathPattern: path, keyPath: nil, statusCodes: statusCodes)
    
    //RKObjectManager objectManager;
    
    func setup(_ baseUrl: String?, path: String?) {
        baseURL = URL(string: baseUrl ?? "")!
        client = AFRKHTTPClient(baseURL: baseURL)
        client.setDefaultHeader("Authorization", value: AUTH_HEADER)
        objectManager = RKObjectManager(httpClient: client)
        let mapping = mapComplexTypes()!
        statusCodes = RKStatusCodeIndexSetForClass(RKStatusCodeClass.successful)
        resDescriptor = RKResponseDescriptor(mapping: mapping, method: RKRequestMethod.GET, pathPattern: path, keyPath: nil, statusCodes: statusCodes)
        objectManager.addResponseDescriptor(resDescriptor)
    }
    /*
    func execute(_ path: String?) {
        let queryParams = [
            "dummy": "dummyParam"
        ]
        //RKObjectManager.shared().getObjectsAtPath(path /*@"/v2/venues/search" */, parameters: queryParams,
        RKObjectManager.shared().getObjectsAtPath("/posts/1", parameters: nil,
                                                  success: { (operation, mappingResult) -> Void in
                                                  //success: { operation, mappingResult in
                                                    self.testUsers = mappingResult?.array();//?.array
        }, failure: { operation, error in
            if let error = error {
                print("Failure of GET:/user with error \(error).")
            }
        })
        
    }
 */
}
